src/runtime_src/core/pcie/linux/pcidev.h

```diff
@@ -142,43 +129,93 @@ class pci_device
+
+  virtual int
+  open(const std::string& subdev, uint32_t idx, int flag) const;
+
+  virtual void
+  close(int devhdl) const;
+
+  virtual int
+  ioctl(int devhdl, unsigned long cmd, void *arg = nullptr) const;
+
+  virtual int
+  poll(int devhdl, short events, int timeoutMilliSec);
+
+  virtual void
+  *mmap(int devhdl, size_t len, int prot, int flags, off_t offset);
+
+  virtual int
+  munmap(int devhdl, void* addr, size_t len);
+
+  virtual int
+  flock(int devhdl, int op);
+
+  virtual int
+  get_partinfo(std::vector<std::string>& info, void *blob = nullptr);
+
+  virtual std::shared_ptr<dev>
+  lookup_peer_dev();
+  
+  // Hand out a "device" instance that is specific to this type of device.
+  // Caller will use this device to access device specific implementation of ishim.
+  virtual std::shared_ptr<device>
+  create_device(device::handle_type handle, device::id_type id) const;
+
+  // Hand out an opaque "shim" handle that is specific to this type of device.
+  // On legacy Alveo device, this handle can be used to lookup a device instance and
+  // make xcl HAL API calls.
+  // On new platforms, this handle can only be used to look up a device. HAL API calls
+  // through it are not supported any more.
+  virtual device::handle_type
+  create_shim(device::id_type id) const;
 
 private:
-  int map_usr_bar(void);
+  int
+  map_usr_bar();
+
   std::mutex lock;
   char *user_bar_map = reinterpret_cast<char *>(MAP_FAILED);
-  bool mgmt = false;
 };
 
-void rescan(void);
-size_t get_dev_total(bool user = true);
-size_t get_dev_ready(bool user = true);
-std::shared_ptr<pci_device> get_dev(unsigned index, bool user = true);
+size_t
+get_dev_total(bool user = true);
+
+size_t
+get_dev_ready(bool user = true);
+
+std::shared_ptr<dev>
+get_dev(unsigned index, bool user = true);
+
+int
+get_axlf_section(const std::string& filename, int kind, std::shared_ptr<char>& buf);
+
+int
+get_uuids(std::shared_ptr<char>& dtbbuf, std::vector<std::string>& uuids);
+
+std::shared_ptr<dev>
+lookup_user_dev(std::shared_ptr<dev> mgmt_dev);
+
+int
+shutdown(std::shared_ptr<dev> mgmt_dev, bool remove_user = false, bool remove_mgmt = false);
 
-int get_axlf_section(const std::string& filename, int kind, std::shared_ptr<char>& buf);
-int get_uuids(std::shared_ptr<char>& dtbbuf, std::vector<std::string>& uuids);
-std::shared_ptr<pcidev::pci_device> lookup_user_dev(std::shared_ptr<pcidev::pci_device> mgmt_dev);
-int shutdown(std::shared_ptr<pcidev::pci_device> mgmt_dev, bool remove_user = false, bool remove_mgmt = false);
-int check_p2p_config(const std::shared_ptr<pcidev::pci_device>& dev, std::string &err);
+int
+check_p2p_config(const std::shared_ptr<dev>& dev, std::string &err);
 
-} // namespace pcidev
+} } // namespace xrt_core :: pci
 
 // For print out per device info
 std::ostream&
-operator<< (std::ostream& stream, const std::shared_ptr<pcidev::pci_device>& dev);
+operator<< (std::ostream& stream, const std::shared_ptr<xrt_core::pci::dev>& dev);
```

Comment:
Unless ownership is shared within this function, passing const std::shared_ptr is not necessary.  It looks like this function should take `const xrt_core::pci::dev*` and that caller should `.get()` and pass the pointer.

Reply:
Ancient code, prefer not to touch.